torch/fx/experimental/symbolic_shapes.py

```diff
@@ -170,6 +170,41 @@ def is_concrete_int(a: Union[int, SymInt]):
 def tensor_has_hints(t):
     return all(has_hint(s) for s in t.size())
 
+def free_symbols(val: Union[SymInt, torch.Tensor]) -> Set[sympy.Symbol]:
+    if isinstance(val, SymInt):
+        return val.node.expr.free_symbols
+    elif isinstance(val, int):
+        return set()
+    elif isinstance(val, torch.Tensor):
+        r = set()
+        for s in val.size():
+            r |= free_symbols(s)
+        for s in val.stride():
+            r |= free_symbols(s)
+        r |= free_symbols(val.storage_offset())
+        return r
```

Comment:
Do you care if these are not symbolic? I know you handle int up there, but is there ever a case you want this to work on just a regular tensor of regular int sizes/strides?

Reply:
I mean, if it's a regular tensor, free_symbols will always return empty set.